File: refresh_browser.c
#include <stdio.h>
#include <curl/curl.h>
#include <unistd.h>

int main(void) {
    CURL *curl;
    CURLcode res;

    curl_global_init(CURL_GLOBAL_DEFAULT);
    curl = curl_easy_init();

    if(curl) {
        while (1) {
            curl_easy_setopt(curl, CURLOPT_URL, "https://example.com");
            res = curl_easy_perform(curl);

            if(res != CURLE_OK)
                fprintf(stderr, "curl_easy_perform() failed: %s\n", curl_easy_strerror(res));
            else
                printf("Page reloaded\n");

            sleep(300);  // 5分間スリープ
        }
        curl_easy_cleanup(curl);
    }

    curl_global_cleanup();
    return 0;
}

File: refresh_browser.rb
require 'curb'

loop do
  begin
    response = Curl.get("https://example.com") #任意のURLを設定する
    if response.status.include?("200")
      puts "Page reloaded"
    else
      puts "Failed to reload page: #{response.status}"
    end
  rescue Curl::Err::CurlError => e
    puts "Curl error: #{e.message}"
  end

  # スリープ時間
  sleep(300)
end
